[PR] Display

philcable opened the pull request:
A shortcode for a front-end view very similar to those found on https://screens.wsu.edu/alumniwallofhonor/.

Commit:
Introduce a shortcode for displaying awardees

Commit:
Enqueue a stylesheet for the shortcode display

This commit also moves the existing `admin.css` file and adds
stylesheet processes to the Grunt build.

Commit:
Enqueue JavaScript with the shortcode

The added JavaScript duplicates the current behaviors on
https://screens.wsu.edu/alumniwallofhonor/.

This also adds processing of the JavaScript to the Grunt build and
the respective dependencies to `package.json`.

Commit:
Update Travis configuration to use stable node and PHP 7.0

philcable commented:
@jeremyfelt #reviewmerge?

jeremyfelt commented on includes/class-wsuwp-alumni-awards.php:
```diff
@@ -271,4 +272,142 @@ public function admin_enqueue_scripts( $hook_suffix ) {
+			return '<!-- No award_slug attribute set -->';
+		}
+
+		wp_enqueue_style( 'alumni-awards', plugins_url( 'css/shortcode.css', dirname( __FILE__ ) ) );
```
I'd add a version here to make it easier to break cache in the future if necessary.

jeremyfelt commented on includes/class-wsuwp-alumni-awards.php:
```diff
@@ -271,4 +272,142 @@ public function admin_enqueue_scripts( $hook_suffix ) {
+		}
+
+		wp_enqueue_style( 'alumni-awards', plugins_url( 'css/shortcode.css', dirname( __FILE__ ) ) );
+		wp_enqueue_script( 'alumni-awards', plugins_url( 'js/shortcode.min.js', dirname( __FILE__ ) ), array( 'jquery' ) );
```
Version here as well (see above).

jeremyfelt reviewed (changes requested):
Looks good, just one quick change. :)

Commit:
Add a version number for registered scripts and styles

This also sets the `$in_footer` parameter to `true` for the
Javascript file enqueue - figured it wouldn't hurt!

philcable commented:
@jeremyfelt Excellent suggestion! Updated - #reviewmerge?

jeremyfelt commented:
:zap:
